### PR TITLE
[8.2.x] Rolling upgrade backports

### PR DESCRIPTION
**Pull request by gustavonalle**
https://issues.jboss.org/browse/ISPN-6539
https://issues.jboss.org/browse/ISPN-7619
https://issues.jboss.org/browse/ISPN-7622
https://issues.jboss.org/browse/ISPN-7584

### CODE REVIEW

**Comment by gustavonalle** [comment]
Failures not related

**Comment by slaskawi** [comment]
@vjuranek I'm not an expert in this.. may I get your opinion here?

File: persistence/remote/src/test/java/org/infinispan/persistence/remote/upgrade/Cluster.java [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+import org.infinispan.manager.EmbeddedCacheManager;
+import org.infinispan.persistence.remote.configuration.RemoteStoreConfigurationBuilder;
+import org.infinispan.server.hotrod.HotRodServer;
+import org.infinispan.server.hotrod.test.HotRodTestingUtil;
+import org.infinispan.test.TestingUtil;
+import org.infinispan.upgrade.RollingUpgradeManager;
+
+
+public class Cluster {
+
```
Comment (vjuranek) [inline persistence/remote/src/test/java/org/infinispan/persistence/remote/upgrade/Cluster.java]: This class is unused and duplicates `TestCluster`

**Comment by vjuranek** [comment]
@slaskawi well, I don't definitely consider myself as an expert on rolling upgrades as well, but if you want my opinion, LGTM (except one minor thing with unused class)

**Comment by vjuranek** [comment]
and one more thing, `8.2.x` branch is now broken by 567769c105cb9778c8623148a74580e18e154bc3, so maybe I'd wait with integration until it's fixed (and tested after rebase - the fix very likely won't affect anything in this PR, but one never knows)

**Comment by gustavonalle** [comment]
@vjuranek @slaskawi The branch is also broken because of the upgrade of JBoss marshalling https://github.com/infinispan/infinispan/pull/4709
/cc @galderz

**Comment by galderz** [comment]
@gustavonalle I looked into it and passed it on to @tristantarrant since it's all related to modules. I have a fix for it but it's not the correct one as mentioned in [ISPN-7652](https://issues.jboss.org/browse/ISPN-7652).

**Comment by galderz** [comment]
Btw, the fix for 8.2.x is [here](https://github.com/galderz/infinispan/commit/cae0926ebe74ad27637f71313a0c1cea5e95e749).

**Comment by tristantarrant** [comment]
I have pushed a mini fix to make 8.2.x compile. @galderz can you make that into a PR ?

**Comment by tristantarrant** [comment]
Merged, thanks @gustavonalle

**Pull request merged**